java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -594,12 +594,73 @@ public final ColumnVector subVector(int start, int end) {
    * @return A new ColumnVector array with slices from the original ColumnVector
    */
   public final ColumnVector[] split(int... indices) {
+    ColumnView[] views = splitAsViews(indices);
+    try {
+      ColumnVector[] columnVectors = new ColumnVector[views.length];
+      for (int i = 0; i < views.length; i++) {
+        columnVectors[i] = views[i].copyToColumnVector();
```

Comment:
This could leak column vectors if a copy fails during the loop.

Reply:
Another way to approach this that may be simpler overall is to keep the native implementation of `split` and add a new native implementation for `splitAsViews`, since the handling of exceptions is simpler in C++ due to the RAII patterns available there.  I'm totally fine if you'd rather keep this transformation in Java instead.